src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.targets

```diff
@@ -153,4 +153,21 @@ Copyright (c) .NET Foundation. All rights reserved.
     <NETSdkWarning ResourceName="UnnecessaryWindowsDesktopSDK" />
   </Target>
 
+  <!-- Default the value of PublishRelease for 8.0+ TFMs. Note that this is pre-evaluated by the CLI in an evaluation before evaluation (see ReleasePropertyProjectLocator.cs).-->
+  <PropertyGroup Condition="'$(PublishRelease)' == '' and '$(_TargetFrameworkVersionWithoutV)' >= '8.0'">
+    <PublishRelease>true</PublishRelease>
```

Comment:
If we want the CLI to handle the case where you have a solution with a .NET 7 and .NET 8 project and neither explicitly sets `PublishRelease`, we need a way for the CLI to detect whether `PublishRelease` was explicitly set.  You could either move the defaulting logic to the CLI, or you could set a property such as `_UsingDefaultPublishRelease` to true here.  I'd probably go with the second option.

Reply:
As discussed offline, this occurs in the pre-evaluation, so it is not necessary.